src/lib/proof-system/zkprogram.ts

```diff
@@ -544,21 +543,14 @@ function isDynamicProof(
   return typeof type === 'function' && type.prototype instanceof DynamicProof;
 }
 
-function getPreviousProofsForProver(
-  methodArgs: any[],
-  { args }: MethodInterface
-) {
-  let proofs: ProofBase[] = [];
-  for (let i = 0; i < args.length; i++) {
-    if (args[i].isProof) proofs.push(methodArgs[i].proof);
-  }
-  return proofs;
+function getPreviousProofsForProver(methodArgs: any[]) {
+  return methodArgs.flatMap(extractProofs).map((proof) => proof.proof);
 }
 
 type MethodInterface = {
   methodName: string;
-  args: { type: ProvableType<unknown>; isProof: boolean }[];
-  proofs: Subclass<typeof ProofBase>[];
+  args: ProvableType<unknown>[];
+  numberOfProofs: number;
   returnType?: Provable<any>;
```

Comment:
simplified the `MethodInterface` type, there was no real use for storing the proofs in the updated code (they have to be deeply extracted from input values anyway)